new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/IstioUtils.java

```diff
@@ -0,0 +1,271 @@
+    getSecureIngressPort.append("-o jsonpath='{.spec.ports[?(@.name==\"https\")].nodePort}'");
+    logger.info("getSecureIngressPort: kubectl command {0}", new String(getSecureIngressPort));
+    try {
+      result = exec(new String(getSecureIngressPort), true);
+    } catch (Exception ex) {
+      logger.info("Exception in getSecureIngressPort() {0}", ex);
+      return 0;
+    }
+    logger.info("getSecureIngressPort: kubectl returned {0}", result.toString());
+    if (result.stdout() == null) {
+      return 0;
+    } else {
+      return new Integer(result.stdout());
+    }
+  }
+
+  /**
+   * Get the tcp ingress port of istio installation.
+   * @returns tcp ingress port for istio-ingressgateway
+   */
+  public static int getIstioTcpIngressPort() {
+    ExecResult result = null;
+    StringBuffer getTcpIngressPort = null;
+    getTcpIngressPort = new StringBuffer("kubectl -n istio-system get service istio-ingressgateway ");
+    getTcpIngressPort.append("-o jsonpath='{.spec.ports[?(@.name==\"tcp\")].nodePort}'");
+    logger.info("getTcpIngressPort: kubectl command {0}", new String(getTcpIngressPort));
+    try {
+      result = exec(new String(getTcpIngressPort), true);
+    } catch (Exception ex) {
+      logger.info("Exception in getTcpIngressPort() {0}", ex);
+      return 0;
+    }
+    logger.info("getTcpIngressPort: kubectl returned {0}", result.toString());
+    if (result.stdout() == null) {
+      return 0;
+    } else {
+      return new Integer(result.stdout());
+    }
+  }
+
+  /**
+   * Deploy the Http Istio Gateway and Istio Virtualservice.
+   * @param domainNamespace name of the domain namespace
+   * @param domainUid name of the domain identifier
+   * @param clusterServiceName  name of WebLogic cluster
+   * @param adminServiceName name of the admin service
+   * @returns true if deployment is success otherwise false
+  */
+  public static boolean deployHttpIstioGatewayAndVirtualservice(
+       String domainNamespace, String domainUid, 
+       String adminServiceName, String clusterServiceName) throws IOException {
+
+    logger.info("Create a staging location for istio configuration objects");
+    Path fileTemp = Paths.get(RESULTS_ROOT, "tmp", "istio");
+    deleteDirectory(fileTemp.toFile());
+    Files.createDirectories(fileTemp);
+
+    logger.info("Copy istio-http-template.service.yaml to staging location");
+    Path srcFile = Paths.get(RESOURCE_DIR, "istio", "istio-http-template.service.yaml");
+    Path targetFile = Paths.get(fileTemp.toString(), "istio-http-service.yaml");
+    Files.copy(srcFile, targetFile, StandardCopyOption.REPLACE_EXISTING);
+
+    String out = targetFile.toString();
+    replaceStringInFile(out, "NAMESPACE", domainNamespace); 
+    replaceStringInFile(out, "DUID", domainUid); 
+    replaceStringInFile(out, "ADMIN_SERVICE", adminServiceName); 
+    replaceStringInFile(out, "CLUSTER_SERVICE", clusterServiceName); 
```

Comment:
Replacing values in template file should be separate from deploying the service since this template is specific to your test. Also, the new file should not be in the src directory - better to put it in pv-root - maybe.

Reply:
Now the signature has been modified 

  /**
   * Deploy the tcp Istio Gateway and Istio virtual service.
   * @param inputTemplateFile input template file
   * @param outputTemplateFile output template file
   * @param templateMap map containing template variable(s) to be replaced
   * @return true if deployment is success otherwise false
  */
  public static boolean deployTcpIstioGatewayAndVirtualservice(
       String inputTemplateFile, String outputTemplateFile,
       Map<String, String> templateMap) throws IOException {

Reply:
I think the calling method should only give the output file name and this method should create it in RESULTS_ROOT/testname. My preference is to have a separate method in utils to copy and update the yaml file (which could be a generic method if other tests want to use it) and that should return the Path where the new file was created which can then be an argument for this deploy method.